cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -12,27 +12,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import AbstractSet, Dict, Iterable, Union, TYPE_CHECKING
+from typing import AbstractSet, Dict, Iterable, Union, TYPE_CHECKING, Sequence, Iterator
 
 import sympy
 
 from cirq import value, protocols
-from cirq._compat import proper_repr
+from cirq._compat import proper_repr, deprecated
 from cirq.ops import (
     raw_types,
     common_gates,
+    gate_operation,
+    dense_pauli_string as dps,
     pauli_string as ps,
     pauli_gates,
     op_tree,
-    pauli_string_raw_types,
+    clifford_gate,
 )
 
 if TYPE_CHECKING:
     import cirq
 
 
 @value.value_equality(approximate=True)
-class PauliStringPhasor(pauli_string_raw_types.PauliStringGateOperation):
+class PauliStringPhasor(gate_operation.GateOperation):
```

Comment:
It looks to me that the main blocker in completely deprecating `PauliStringPhasor` operation and replacing all uses with `PauliStringPhaseGate` + the automatic gate operation would be 
- A bunch of convenience properties like `exponent_neg`, `exponent_pos` etc.; which currently exist on the operation and would later exist on the gate. 
- `pass_operations_over` method which modifies the underlying pauli string and hence would need to exist on the operation and not on the gate. 


I see very few uses of the `pass_operations_over` method in cirq and would argue in favour of deprecating it from the operation. Ideally, I would only want to have `PauliStringPhasorGate` and deprecate `PauliStringPhasor` operation. Let's discuss this again in the cirq sync today to see what others think about it.

Reply:
I think that would be best, but also that it should be a separate PR.

Most of what breaks if we remove PauliStringPhasor is in contrib/pauli_string. I'm pretty sure everything can be patched up nicely to work with gate+qubits, but there's a fair amount of patching to do. Whereas the current PR is self-contained.

Reply:
Another alternative could be to define a private `_PauliStringPhasorOperation` which extends `GateOperation` and defines the additional convenience methods like `pass_operations_over`. 

What I do not want to is to continue supporting both public APIs:  `PauliStringPhasor` (an operation) and `PauliStringPhasorGate` (a gate). The two other places where we currently have this structure (i.e. both gate and operations for the same item are part of the public API) is `ControlledGate` & `ControlledOperation` and `DensePauliString` and `PauliString`. 

I'd like to avoid creating more such cases unless we see a strong use case. LMK your thoughts on this.

Reply:
I'm in favor of deprecating and I'd strongly prefer deprecating it rather than creating a private thing that inherits GateOperation. But still, as a subsequent PR to this one.

Alternatives to consider for `pass_operations_over`: we could plop a `pass_operations_over` onto GateOperation, we could make it a protocol, or we could just move that logic to `contrib/pauli_string` rather than being part of the gate/operation itself.

Reply:
Per cirq sync, we agreed it's best to deprecate `PauliStringPhasor` as a separate PR, so we can consider an approach with this and `PauliString` at the same time.